IanGRap/Squish
# Win screen

Contributors: Austin

Goal:
- Create a more interesting win screen

What Changed
- When I started writing code, I realized this feature would be a lot easier if I did a refactor.
    - As a result, I refactored all of the private inner classes storing information about players like their controller index and material in a new public class called PlayerData. It replaced the classes called PlayerInfo, Player and RespawnBlockData
- New Files:
    - WinScreenController.cs
    - WinScreenCameraController.cs
- Renamed files:
    - UIGameOverMenu --> UIWinScreenMenu
- New Scene:
    - Win Scene
- Important Alterations:
    - UIMatchManager
        - No longer enables a canvas when the game ends. Instead, it starts the new scene.
    - MatchManager
        - Now records player time of death so that it can be used to order the players in the win screen.

How to Test:
- Load up the match scene and play some games.
- Make sure the win screen loads and shows the correct order of players. The players should be ordered first by their scores and second by their time of death in the last round of the game.
- Make sure the camera transition plays when you click on UI elements within the win screen
- Make sure the feature works with 2, 3 and 4 players.
    - NOTE: I did not get to test this because I was unable to use controllers while making the feature
- Make sure it works in a build
- Let me know how you think it looks :)
